lms/djangoapps/course_api/api.py

```diff
@@ -227,9 +229,18 @@ def get_due_dates(request, course_key, user):
                 url: the deep link to the block
                 date: the due date for the block
     """
+    try:
+        outline = get_course_outline(course_key)
```

Comment:
Why use `get_course_outline` here rather than `get_user_course_outline`?

Reply:
FWIW, this is still going to be reasonably fast (the median time for this call in the past week was 7 ms). At some point I think we should make a more centralized accounting of the active published version for various apps, because we have so many things that are building their own post-publish model of the world... but that's for a different day. 😛